### PR TITLE
Sitemap fixes

### PR DESCRIPTION
**Pull request by dougjohnston**
A largely rewritten sitemap view with a focus on performance. Sitemap data is now serialized in the model and fed to the view for display. Extraneous partial calls were removed, with most view code living in a single partial for performance reasons.

**Caching**
Caching no longer seems necessary and has been removed entirely. Load times are now in the 5-6s range which seems more than acceptable for a page that used to regularly time out.

**Drag and Drop**
Use of the existing jQuery UI drag/drop libraries was eventually abandoned for being too sluggish. Instead, uses HTML5 native support for drag and drop, which is actually quite nice (limited in features, but great for our purposes).

**Testing**
Best to test this locally. See CMS PR for details.

**Code Quality Note**
FWIW, if I were to do this again, I'd probably rip out all of the old sitemap code and rebuild it from scratch. But I initially wanted to change as little as possible, so this became a series of **many** small iterations that eventually morphed into its present state. So, it's not quite as clean as I would like.

### CODE REVIEW

File: app/assets/javascripts/cms/ajax.js [hunk 1]
```diff
@@ -55,6 +55,16 @@ jQuery(function ($) {
         beforeSend: $.cms_ajax.asJSON()
     });
     $(document).ajaxError(function (x, status, error) {
-        alert("A " + x.status + " error occurred: " + error);
+        errorMessage = 'AJAX Error (status = ' + status.status + '):\n'
```
Comment (allynfolksjr) [inline app/assets/javascripts/cms/ajax.js:58]: Is Rollbar set up for CMS JS? Maybe it's time... :|

File: app/models/cms/section.rb [hunk 2]
```diff
@@ -92,10 +92,56 @@ def pages
       child_pages.compact
     end
 
-    def self.sitemap
+    def self.old_sitemap
```
Comment (allynfolksjr) [inline app/models/cms/section.rb]: `self.garbage_sitemap`

File: app/models/cms/section.rb [hunk 3]
```diff
@@ -92,10 +92,56 @@ def pages
...
+    def self.sitemap
+      if ENV['CACHE_SITEMAP']
+        Rails.cache.fetch(["sitemap-#{sitemap_contents_last_updated}"]) { build_sitemap }
+      else
+        build_sitemap
+      end
+    end
+
+    # Build the sitemap hash
+    def self.build_sitemap
```
Comment (allynfolksjr) [inline app/models/cms/section.rb:107]: Clever approach!

File: app/models/cms/section_node.rb [hunk 4]
```diff
@@ -87,10 +87,28 @@ def deletable?
     true
   end
 
+  def icon_style(child_count)
```
Comment (allynfolksjr) [inline app/models/cms/section_node.rb:91]: TIL CMS has built-in icon styles

**Review by allynfolksjr — APPROVED** [review]
Nice work with wrestling this monster down to a more manageable level! Moving a lot of the code to the view-level is a clever solution and keeps it sane as the sitemap continues to grow in size.

File: app/assets/javascripts/cms/ajax.js (hunk 1 above)
Comment (dougjohnston) [reply app/assets/javascripts/cms/ajax.js:58]: Done!

**Review by bill-ricksteves — APPROVED** [review]

**Pull request merged**